1_fetch/src/get_nwis_data.R

```diff
@@ -1,48 +1,34 @@
+### Create a function to pull down individual sites because our website
+### is 'glitchy'.  This avoids repulling everything if an individual site fails
 
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000", "01436690", "01466500")){
+download_nwis_data <- function(site_num, parameterCd = '00010', 
+                               startDate="2014-05-01", endDate="2015-05-01"){
   
-  # create the file names that are needed for download_nwis_site_data
-  # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
-  # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
-  data_out <- data.frame()
-  # loop through files to download 
-  for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
-    # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
-    data_out <- bind_rows(data_out, these_data)
-  }
-  return(data_out)
-}
-
-nwis_site_info <- function(fileout, site_data){
-  site_no <- unique(site_data$site_no)
-  site_info <- dataRetrieval::readNWISsite(site_no)
-  write_csv(site_info, fileout)
-  return(fileout)
-}
-
-
-download_nwis_site_data <- function(filepath, parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01"){
-  
-  # filepaths look something like directory/nwis_01432160_data.csv,
-  # remove the directory with basename() and extract the 01432160 with the regular expression match
-  site_num <- basename(filepath) %>% 
-    stringr::str_extract(pattern = "(?:[0-9]+)")
+  filepath <- file.path('1_fetch/tmp',paste0(site_num,'.csv'))
```

Comment:
With targets, we usually want to keep filepath information at the top-level (in  `_targets.R`) because it makes it easier for others (and you some time in the future when you haven't used the pipeline for awhile) to use the pipeline so that they don't need to go hunting through scripts to find where a file was saved. I would recommend following how `download_nwis_site_data()` was originally set up, where you pass in the `filepath` and then extract the site number using `basename(filepath) %>% stringr::str_extract(pattern = "(?:[0-9]+)")`